File: packages/python/examples/pytest/drag_test.py
from os import getenv

from pytest import mark

from alumnium import Alumni
from alumnium.tools import DragTool

driver_type = getenv("ALUMNIUM_DRIVER", "selenium")


@mark.xfail("appium" in driver_type, reason="Drag is not implemented in Appium yet")
def test_drag(al, driver, navigate):
    al = Alumni(
        driver,
        extra_tools=[DragTool],
    )

    navigate("slider.html")

    al.do("drag the slider to the right by 10 pixels")
    al.check("the slider value is greater than 50")

    al.do("drag the slider to the left by 10 pixels")
    al.check("the slider value is less than or equal to 50")

File: packages/typescript/examples/mocha/drag.test.ts
import { Alumni } from "../../src/Alumni.js";
import { DragTool } from "../../src/tools/DragTool.js";
import "./globals.js";
import { navigate } from "./helpers.js";

describe("Drag", () => {
  it("moves element", async function () {
    const al = new Alumni(driver, {
      extraTools: [DragTool],
    });

    await navigate(driver, "slider.html");

    await al.do("drag the slider to the right by 10 pixels");
    await al.check("the slider value is greater than 50");

    await al.do("drag the slider to the left by 10 pixels");
    await al.check("the slider value is less than or equal to 50");
  });
});
